### PR TITLE
Use token authentication for Tower

### PR DESCRIPTION
**Pull request by mshriver**
Support token authentication for AnsibleTower provider.

On the tower side, this requires an Application, which the tokens are registered under, and a global setting to allow token creation for OAuth users.

When using token, username and password settings are no longer necessary.

### CODE REVIEW

File: broker/providers/ansible_tower.py [hunk 1]
```diff
@@ -25,14 +27,28 @@ def __init__(self, **kwargs):
         self._construct_params = []
         config = kwargs.get("config", awxkit.config)
         config.base_url = AT_URL
-        config.credentials = {"default": {"username": UNAME, "password": PWORD}}
-        config.use_sessions = True
-        if "root" in kwargs:
```
Comment (mshriver) [inline broker/providers/ansible_tower.py:30]: @JacobCallahan looking for your input here - what was the intent of support `root` instance being passed into this constructor?

Comment (mshriver) [reply broker/providers/ansible_tower.py:30]: Nvm, found it, the mock stub behavior for unit tests.

File: broker/providers/ansible_tower.py [hunk 2]
```diff
@@ -13,8 +14,9 @@
 from broker import helpers
 
 AT_URL = settings.ANSIBLETOWER.base_url
-UNAME = settings.ANSIBLETOWER.username
-PWORD = settings.ANSIBLETOWER.password
+UNAME = settings.ANSIBLETOWER.username.get("username")
```
Comment (mshriver) [inline broker/providers/ansible_tower.py:17]: This is a temporary implementation of safety around reading these settings, until we've got better support for validators and defaults in DynaConf.

Comment (JacobCallahan) [reply broker/providers/ansible_tower.py:17]: We may get a new dynaconf version soon, and this would end up unnecessary if handled by an upcoming pr

Comment (mshriver) [reply broker/providers/ansible_tower.py:17]: I'm happy to rebase after that.

Comment (mshriver) [reply broker/providers/ansible_tower.py]: Moved to validators - these `.get()` calls are still necessary though so this module has safe import.

File: broker/settings.py [hunk 3]
```diff
@@ -29,6 +27,13 @@
     Validator("HOST_USERNAME", default="root"),
 ]
 settings = Dynaconf(settings_file=str(settings_path), validators=validators,)
+
+# Handle both modes of ansible tower authentication
+settings.validators.register(
+    ((Validator("ANSIBLETOWER.username", must_exist=True) & Validator("ANSIBLETOWER.password", must_exist=True))
+    | Validator("ANSIBLETOWER.token", must_exist=True))
+)
```
Comment (JacobCallahan) [inline broker/settings.py:35]: can this be combined with the above validators and included during the settings initialization?

Comment (mshriver) [reply broker/settings.py]: yep! Didn't think I'd be able to include it as a list item, but it works.

**Review by JacobCallahan — APPROVED** [review]
ACK

**Pull request merged**